### DISCUSSION TITLE
Good practice for env variables in Quasar v1, now that qenv / dotenv extensions are deprecated?

### DISCUSSION THREAD

**Discussion by davthedev** [quasarframework/quasar#9318] [Quasar App Extensions]
Hi

We had the dotenv / qenv extensions recommended to manage environment variables. They are deprecated since Apr 7.
It is noticed to use the env section of the quasar configuration file instead.

But I need to store environment-dependent variables, including API keys, which should not be present in version control. The quasar.conf.js file is not the right place for such information.

I was referenced to look at the updated doc in Quasar Next documentation, that directly recommends using the "real" dotenv, instead of the extensions. https://next.quasar.dev/quasar-cli/handling-process-env#using-dotenv
This one is related to v2, which is still in beta.

The situation is a bit confusing. I am stuck choosing between a deprecated solution (quasar-dotenv / qenv) and a beta one (v2). So I went on the quasar-dotenv route. It seems to work for now.

For a project that is due to go to production using the stable Quasar v1, which best practice you would recommend to manage environment variables today?

**Comment by hawkeye64** [comment]
@davthedev In the end it's just JavaScript. There are many ways you can do it. I would just write a script, then import that into quasar.conf.js and then apply it to `build -> env`.

**Comment by dany-eudes** [comment]
Maybe that would be useful:

Install dotenv package:
https://www.npmjs.com/package/dotenv


* ./config/envparser.js

```` javascript
const DotEnv = require('dotenv')
const parsedEnv = DotEnv.config().parsed

module.exports = function () {
  // Let's stringify our variables
  for (key in parsedEnv) {
    if (typeof parsedEnv[key] === 'string') {
      parsedEnv[key] = JSON.stringify(parsedEnv[key])
    }
  }
  return parsedEnv
}
```` 
* quasar.conf.js 
``` javascript

   const envparser = require('./config/envparser')
   (...)

    build: {
      (...)
      env: envparser(), 
      (...)   
    }
```

Somewhere, in your code, test it with a .env variable:
``` javascript
console.log( process.env.MY_ENV_VAR );    
```

**Reply by hawkeye64** [reply]
If you need server variables, look at the code here and extrapolate it into your own code: https://github.com/quasarframework/app-extension-dotenv/blob/dev/app-extension/src/index.js

**Reply by hawkeye64** [reply]
Specifically this: https://github.com/quasarframework/app-extension-dotenv/blob/dev/app-extension/src/index.js#L12
